### PR TITLE
Subscribe and Unsubscribe for Associations

### PR DESCRIPTION
**Pull request by Adriv020**
In this PR I implemented the logic for subscirbing and unsubscribing for association, addressing issue [#192](https://github.com/EPFL-Life/life/issues/192).

Details:
- I've created 2 new methods in **UserRepository**: **subscribeToAssociation** and **unsubscribeFromAssociation**.
- These two methods have been implemented in both **UserRepositoryLocal** and **UserRespoitoryFirestore** classes
- I've created their respective tests in **UserRepositoryLocalTest** and **UserRespoitoryFirestoreTest**.
- In UserRespoitoryFirestore and UserRespoitoryLocal we repated the same message while getting the current user multiple times, fact that causes Sonar issues. For addressing this problem I created a constant with the message in UserErrorConstants.kt. I've considered define it as a string in R.string but this causes problems with the synchronization in the Gradle build.

TODO:
The logic works and the database is updated correctly when the user subscribes or unsubscribes for an event. Now we need to visualize these changes in the UI, task that @pukoa420 is going to do ([#244](https://github.com/EPFL-Life/life/issues/244))

### CODE REVIEW

File: app/src/main/java/ch/epfllife/model/user/UserRepositoryFirestore.kt [hunk 1]
```diff
@@ -197,4 +197,64 @@ class UserRepositoryFirestore(private val db: FirebaseFirestore) : UserRepositor
...
+      return Result.failure(
+          NoSuchElementException(
+              "Association with ID $associationId does not exist in the repository."))
+    }
+
+    // case 3: the user is trying to unsubscribe from an association they are not subscribed to
+    if (!currentUser.subscriptions.contains(associationId)) {
+      return Result.failure(
+          IllegalArgumentException(
+              "User is already subscribed to association with ID: $associationId"))
```
Comment (pukoa420) [inline app/src/main/java/ch/epfllife/model/user/UserRepositoryFirestore.kt]: This error message is misleading, it should be somthing like "User is not subscribed to association with ID: $associationId""

File: app/src/main/java/ch/epfllife/model/user/UserErrorConstants.kt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package ch.epfllife.model.user
+
+// for now we will use const messages for avoiding sonar issues
+const val ERROR_USER_NOT_LOGGED_IN = "No user is currently logged in"
```
Comment (pukoa420) [inline app/src/main/java/ch/epfllife/model/user/UserErrorConstants.kt]: I like the idea behind this, but i would consider moving this constant to the classes (even if that means a small duplicate) or put it into strings.xml and you the Resource reference.

**Review by pukoa420 — CHANGES_REQUESTED** [review]
Good implementation, proper tests -> all edge cases checked.
Please just quickly fix my few comments and then ready to merge
